atom/browser/api/atom_api_session.cc

```diff
@@ -175,6 +177,49 @@ struct Converter<net::ProxyConfig> {
   }
 };
 
+template<>
+struct Converter<history::QueryResults*> {
+
+  static v8::Local<v8::Value> ToV8(v8::Isolate* isolate,
+                                   history::QueryResults *val) {
+
+    v8::Local<v8::Array> entries(MATE_ARRAY_NEW(isolate, static_cast<int>(val->size())));
```

Comment:
you probably want to use `base::ListValue` and `base::DictionaryValue` here

Reply:
OK. I'll look at this. The existing code worked well. I don't remember the justification for using the included types. I'll try the other version.